packages/neuron-wallet/src/services/tx/transaction-persistor.ts

```diff
@@ -30,37 +30,56 @@ export class TransactionPersistor {
     const connection = getConnection()
     const txEntity: TransactionEntity | undefined = await connection
       .getRepository(TransactionEntity)
-      .findOne(transaction.hash, { relations: ['inputs', 'outputs'] })
+      .findOne(transaction.hash)
 
     // return if success
     if (txEntity && txEntity.status === TransactionStatus.Success) {
       return txEntity
     }
 
     if (txEntity) {
+      // lazy load inputs / outputs
+      const inputEntities = await connection
```

Comment:
Please run and see if loading tx, inputs, outputs separately as three queries are actually faster than loading them as a join query `{ relations: ['inputs', 'outputs'] }`.

Reply:
See the code above it, the main reason is only need to load `inputs` and `outputs` if `txEntity.status` not `success`

Reply:
I see. That makes sense.